### PR TITLE
Allow hiding of inactive builders in Waterfall

### PR DESCRIPTION
**Pull request by rajdeepbharati**
Closes https://github.com/buildbot/buildbot/issues/4761

## Contributor Checklist:

* [ ] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

File: www/waterfall_view/src/module/main.module.js [hunk 1]
```diff
@@ -199,6 +209,23 @@ var WaterfallController = (function() {
...
+            if ((builder.masterids == null)) {
+                return false;
+            }
+            for (let mid of Array.from(builder.masterids)) {
+                const m = this.$scope.masters.get(mid);
+                if ((m != null) && m.active) {
+                    active = true;
+                }
+            }
+            if (builder.tags.includes('virtual')) {
```
Comment (tardyp) [inline www/waterfall_view/src/module/main.module.js]: ditto

**Comment by tardyp** [comment]
Shall we pursue this approach?

**Comment by rajdeepbharati** [comment]
@tardyp Did you have other ideas? IMO this is the easy way to handle it.

**Comment by tardyp** [comment]
It is easy, but confusing. I think we should think of a better approach. I think the best would be to show builders that are visible in the current vertical scroll timeframe.
This makes the algorithm more complex, but this would be an interresting algorithm to work-on.
This concept is used for most web text editors, and the logviewer is also implementing it.

**Comment by rajdeepbharati** [comment]
> show builders that are visible in the current vertical scroll timeframe

Doesn't that already happen? I'm a bit confused here (how are these related?)

**Pull request merged**